### ISSUE TITLE
[consistent-type-assertions] Disable check as const if assertionStyle: never

### ISSUE THREAD

**Issue by cevek** [typescript-eslint/typescript-eslint#992]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/consistent-type-assertions": ["error", { "assertionStyle": "never" }]
  }
}
```

```TS
const x = {a: «foo»} as const
```

**Expected Result**
No error
**Actual Result**
error  Do not use any type assertions
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.1` |
| `@typescript-eslint/parser`        | `2.2` |
| `TypeScript`                       | `3.6` |
| `ESLint`                           | `6.3` |
| `node`                             | `12` |
| `npm`                              | `6` |

**Comment by a-tarasyuk** [comment]
@bradzacher 

Is it a bug or need to add a new option to handle `as const` (`<const>`) assertion for `{ "assertionStyle": "never"   }`?

Right now the following code with the option `{ "assertionStyle": "never"   }` reports error, 

```TS
const x = [1] as const;
```

Is it the right behavior? Or need to ignore const assertion for option `{ "assertionStyle": "never"   }` and the following code should be valid?

```TS
// @typescript-eslint/consistent-type-assertions": ["error", { "assertionStyle": "never" }]
const x = [1] as const;
const y = { a: 1 } as const;
```

**Comment by bradzacher** [comment]
I'm still undecided tbh.
Technically this is a "safe" assertion, but it is an assertion none the less.

I can see arguments going both directions, but I think this is probably safe to allow in the rule as a special case, considering it cannot ever weaken any typings.

**Comment by ark120202** [comment]
One problem is that with `"assertionStyle": "never"` there is no way to specify whether `<const>{}` or `{} as const` should be preferred, so that's another point for discussion in #709

**Comment by bradzacher** [comment]
yeah. I really wish there was a compiler-level flag for disabling angle-bracket assertions... I mean there kind is (by renaming the file to `.tsx`), but it seems like it should just be "blockable".

I was thinking maybe we bring back the old rule, or maybe we add more options to this rule to make it more flexible.
It probably requires a bit of a rework / rethink. I don't know how valuable it is to allow linting against angle bracket assertions whilst blocking all assertions - there's only really been one person asking for it.

**Comment by garyking** [comment]
Please allow an option to specifically disable checks for `as const`, even when using `never`.
